src/Language/PureScript/Environment.hs

```diff
@@ -241,6 +241,8 @@ primTypes =
     , (primName "Boolean",  (Star, ExternData))
     , (primName "Partial",  (Star, ExternData))
     , (primName "Fail",     (FunKind Symbol Star, ExternData))
+    , (primName "TypeString", (FunKind Star Symbol, ExternData))
+    , (primName "TypeConcat", (FunKind Symbol Symbol, ExternData))
```

Comment:
OK, I looked up why the kind checker doesn't flag this. It's because the kind checking of the instance happens when we're in typeCheckAll and we reach the TypeClassDictionaryAppConstructor. However, we don't get there because we first check the main function, and hit the "NoInstanceFound" check in typesOf. When you remove the main function, you do get the kinds error, as expected.

The ordering of declarations is still a blind spot for me at the moment, so not sure yet how to fix it.